scrapper.py

```diff
@@ -25,39 +36,159 @@ class Crawler:
+        articles_code_stored_here = article_bs.find_all('div', {'class': 'articles'})
+        for article_code in articles_code_stored_here:
+            links_articles = article_code.find_all('a')
+            for link in links_articles:
+                if self.max_articles == self.count_articles:
+                    break
+                try:
+                    art_link = link["href"]
+                    if art_link.startswith('/') and 'pdf' not in art_link:
+                        self.urls.append('https://periodical.pstgu.ru' + link["href"])
+                        self.count_articles += 1
+                except KeyError:
+                    continue
 
     def find_articles(self):
         """
         Finds articles
-        """
-        pass
+        """  # I also don't understand what I need to do here, because this code can be moved to extract_url
+        for seed_url in self.seed_urls:
+            issue_page = requests.get(seed_url)
+            issue_page_bs = BeautifulSoup(issue_page.text, 'html.parser')
+            self._extract_url(issue_page_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
+
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def parse(self):
+        article_page = requests.get(self.url)
+        article_bs = BeautifulSoup(article_page.text, 'html.parser')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+        list_of_possible_l = ['Список литературы', 'ЛИТЕРАТУРА', 'Список использованных источников и литературы',
+                              'Источники']
+
+        class_of_link = article_bs.find('div', {'class': "articles-item-more"})
+        if class_of_link:
+            link_to_save = class_of_link.find('a')
+            link_to_pdf = 'https://periodical.pstgu.ru' + link_to_save['href']
+            final_page = requests.get(link_to_pdf)
+            final_page_bs = BeautifulSoup(final_page.text, 'html.parser')
+        else:
+            final_page_bs = article_bs
+        url_pattern = re.compile(r"var DEFAULT_URL = '(.*?)'")
+        final_link = final_page_bs.find(string=url_pattern)
+        final_link = 'https://periodical.pstgu.ru' + \
+                     str(final_link).replace('var DEFAULT_URL = ', '').replace("'", '').replace(';', '').strip()
+        raw_pdf = PDFRawFile(final_link, self.article_id)
+        raw_pdf.download()
+        text_from_pdf = raw_pdf.get_text()
+        for refs_maker in list_of_possible_l:
+            if refs_maker in text_from_pdf:
+                text_pdf_list = text_from_pdf.split(refs_maker)
+                del text_pdf_list[-1]
```

Comment:
??

Reply:
I used this part of the code to separate and delete the list of references ( as it was a part of the task).